src/Bitcoin/TransactionBuilder.cpp

```diff
@@ -158,6 +167,16 @@ TransactionPlan TransactionBuilder::plan(const SigningInput& input) {
                 plan.change = 0;
             }
             plan.fee = estimateSegwitFee(feeCalculator, plan, output_size, input);
+
+            // `InputSelector` has a rough segwit fee estimation algorithm,
+            // so the fee could be increased or decreased (see `InputSelector::select`).
+            // We need to make sure if we have enough UTXOs to cover "requested amount + final fee".
+            if (!maxAmount && plan.availableAmount < plan.fee + plan.amount) {
```

Comment:
Shouldn't this also be checked and return an error when `maxAmount == true`?

Reply:
No, because the `plan.amount` will be adjusted according to `plan.availableAmount` below at this line:
https://github.com/trustwallet/wallet-core/blob/ff77450cdcb163f1d997cda15dbeefa477191b15/src/Bitcoin/TransactionBuilder.cpp#L189-L190